htdocs/ticket/class/ticket.class.php

```diff
@@ -570,7 +571,18 @@ public function create($user, $notrigger = 0)
 
 			if (!$error && !empty($conf->global->TICKET_ADD_AUTHOR_AS_CONTACT)) {
 				// add creator as contributor
-				if ($this->add_contact($user->id, 'CONTRIBUTOR', 'internal') < 0) {
+
+				// We first check the type of contact (internal or external)
```

Comment:
I suggest 2 behaviour :
If TICKET_ADD_AUTHOR_AS_CONTACT = 1, we do old method, with 2, we do the one you suggest...

Reply:
I just added the check on constant == 2 to do what I suggested.